src/olympia/reviewers/utils.py

```diff
@@ -113,18 +114,24 @@ class Meta(AddonQueueTable.Meta):
         orderable = True
 
     @classmethod
-    def get_queryset(self, request, *, upcoming_due_date_focus=False, **kw):
-        show_only_upcoming = upcoming_due_date_focus and not acl.action_allowed_for(
-            request.user, amo.permissions.ADDONS_ALL_DUE_DATES
-        )
-        qs = Addon.unfiltered.get_queryset_for_pending_queues(
-            admin_reviewer=is_admin_reviewer(request.user),
-            show_temporarily_delayed=acl.action_allowed_for(
+    def get_queryset(cls, request, *, upcoming_due_date_focus=False, **kwargs):
```

Comment:
nit: is there a reason not to simply define the 3 relevant properties as args and let the view pass in the arguments from else and the cron pass in the hard coded values. IMO that makes it super clear at the call site what is going on, whereas now both are kind of obfuscated.. If there is a good reason to keep the request in context then fine, but otherwise that seems like the better way to go, don't you think?

Reply:
I'm not sure why it needs to be 2 methods. it could just be one method with the defined arguments.

```python
get_queryset(
    admin_reviewer=False,
    show_temporarily_delayed = True,
    show_only_upcoming = False,
)
```

Then in the view we can pull the arguments from the request.

```python
admin_reviewer = is_admin_reviewer(request.user)
show_temporarily_delayed = acl.action_allowed_for(
    request.user, amo.permissions.ADDONS_TRIAGE_DELAYED
)
show_only_upcoming = upcoming_due_date_focus and not acl.action_allowed_for(
    request.user, amo.permissions.ADDONS_ALL_DUE_DATES
)

table = TheClass.get_queryset(
    admin_reviewer=admin_reviewer,
    show_temporarily_delayed=show_temporarily_delayed,
    show_only_upcoming=show_only_upcoming,
)

... do stuff with table
````

And in the cron we just directly set whatever arguments we want... would something like this not work and achieve my goal of keeping the control logic at the call site? You can say yes and also not implement it, but wondering if this would work in your view?

Reply:
Something like that would work, yes, but I was trying to avoid more duplication of code.  The code to check for `show_temporarily_delayed` / `admin_reviewer` is needed in the queue view... but also the dashboard, which fetches the counts (personalized for the current user according to what they'd be able to see). The single `get_queryset()` method avoids that duplication currently.